be/src/exec/hdfs_scanner_text.cpp

```diff
@@ -370,7 +370,9 @@ Status HdfsTextScanner::_create_or_reinit_reader() {
 Status HdfsTextScanner::_get_hive_column_index(const std::string& column_name) {
     for (int i = 0; i < _scanner_params.hive_column_names->size(); i++) {
         const std::string& name = _scanner_params.hive_column_names->at(i);
-        if (name == column_name) {
+        const bool case_sensitive = _scanner_params.case_sensitive;
```

Comment:
The `case_sensitive` parameter is not related to the column index, so it can be moved out of the for-loop to avoid assigning it multiple times.

Reply:
+1.
`_columns_index` is a map, its key is better for saving the origin SlotDescriptor's name, not the formatted name.